rpm/pipy.spec

```diff
@@ -60,8 +60,8 @@ source /opt/rh/rh-nodejs14/enable
 %endif
 cd build
 CXX=clang++ CC=clang cmake3 \
-  -DPIPY_GUI==%{?with_gui:ON}%{?!with_gui:OFF} \
-  -DPIPY_SAMPLES=%{?with_gui:ON}%{?!with_gui:OFF} \
+  -DPIPY_GUI=%{?with_gui:ON}%{!?with_gui:OFF} \
```

Comment:
@nixff hi York, could you help me understand this change? what does it fix? and how? i was referencing https://rpm-software-management.github.io/rpm/manual/macros.html#conditionally-expanded-macros.

Reply:
@tchaikov I think I referred the same link at that moment, and thought the syntax should be `%{!?}` instead of `%{?!}`? then I make this change.

Reply:
@nixff let me quote syntax in the linked document

> %{?!macro_name:value}

i am not sure what your change fixes.

Reply:
oic. the document is not consistent..

Reply:
@nixff anyway, i am still curious what your change fixes. if it's just wrong. we need to fix the document.

Reply:
> @nixff anyway, i am still curious what your change fixes. if it's just wrong. we need to fix the document.

This pr was going to fix the errors, as mentioned in the first comment, a typo and PATH error, which cause the build failure like (https://github.com/flomesh-io/pipy/actions/runs/3629897593).

The macros is irrelavant to the failure, but I submitted together after I referred to the doc and test in local.

Reply:
fair enough, as i still don't think the spec changes are necessary.